---
title: "Calculating mutation load"
---


### Calculating load

There are various types of load and in general, genetic load can be divided between potential and realized load. Realized load (also known as expressed load) only includes the deleterious mutations that are expressed in the individual (Bertorelle et al., 2022). Potential load (also known as inbreeding or masked load) is the fitness reduction due to deleterious mutations, of which not all are expressed on an individual level and therefore quantifies recessive deleterious mutations that could be expressed in future generations (Bertorelle et al., 2022).

I thus calculated the potential load as the number of loci that contained a mutation in heterozygosity divided by the number of variants within that mutation category class that were genotyped in that individual. The realized load was calculated as the number of loci that contained a mutation in homozygosity divided by the number of variants within that mutation category class that were genotyped in that individual. The total genetic load sums the potential and realized load.

## Results

### Number of mutations per impact class

First, let's look at the distribution of the number of mutations per type and impact class that we identified in this population (@fig-hist-category).

![Histograms mutations per type](../plots/mutations_per_type/n_per_type_posternotitle.png){#fig-hist-category}

Here we can see that every LoF mutation is a high impact mutation.

### Number of mutations per scaffold

A total of 261 scaffolds (out of the total 21,979 scaffolds) contain a mutation (either missense, high impact or LOF). You can see the distribution of the number of mutations on a scaffold below (@fig-hist-mutations-scaffold):

![Histogram of the number of mutations on a scaffold](../plots/mutations_per_scaffold.png){#fig-hist-mutations-scaffold}

### Distribution of mutations

I used a sliding window approach to visualise the distribution of mutations across the genome. I only took the 25 largest scaffolds with mutations, which still includes 95% of mutations. The sliding windows were 500kb in size and I summed the number of mutations present in the window for each for each of the mutation types separately. See below the distribution for high impact mutations (@fig-dist-high):

#### Distribution high impact mutations

![Distribution of high impact mutations across the genome](../plots/distribution_mutations_high_acrossgenome.png){#fig-dist-high}

It's interesting that the patterns are very similar across the mutation types, that there are peaks at the edges of the scaffolds which is in line with recombination rates being higher at the ends of chromosomes. There are higher peaks in larger scaffolds, and in general more mutations in larger scaffolds compared to smaller ones.

### Relationship number of mutations and scaffold size

One hypothesis to explain high realised load of lekking males is through genetic hitchhiking of deleterious mutations alongside positively selected sexual traits (QTLs). (@fig-mut-scafsize). Over the course of evolution, selection will favour sexually selected QTLs to move on micro-chromosomes to reduce the impact of this genetic hitchhiking. Although we do not have chromosome-level assembled scaffolds, we can still investigate the relationship between scaffold size and the number of mutations.

![Relationship scaffold size and the number of mutations](../plots/mutations_per_scaffoldsize.png){#fig-mut-scafsize}

It indeed seems like the smaller scaffolds have higher number of mutations, in all mutation classes. This result should be interpreted with care though, as scaffold size is not necessarily representative of chromosome size.

### Genetic load

Then, based on the definitions described above, I calculated realized, potential and total genetic load on an individual basis. We can see the distribution of the different load types below(@fig-hist-load-perid).

![Histogram load per type per ID](../plots/load_per_id.png){#fig-hist-load-perid}