<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.340">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>readme</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="README_files/libs/clipboard/clipboard.min.js"></script>
<script src="README_files/libs/quarto-html/quarto.js"></script>
<script src="README_files/libs/quarto-html/popper.min.js"></script>
<script src="README_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="README_files/libs/quarto-html/anchor.min.js"></script>
<link href="README_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="README_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="README_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="README_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="README_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">



<section id="predicted-deleterious-mutations-reveal-the-genomic-mechanisms-underlying-fitness-variation-in-a-lekking-bird" class="level1">
<h1>Predicted deleterious mutations reveal the genomic mechanisms underlying fitness variation in a lekking bird</h1>
</section>
<section id="general" class="level1">
<h1>General</h1>
<p>This is the repository that contains all scripts used for the analysis in the manuscript titled “Predicted deleterious mutations reveal the genomic mechanisms underlying fitness variation in a lekking bird” by Chen et al.&nbsp;(in preparation). Genomic data, including the reference genome, and the 190 resequenced grouse genomes can be found on NCBI (see links below) and phenotypic data as well as the black grouse annotation are stored in this repository.</p>
<p>Below you will find an explanation of which data files can be found where, and the general structure of the workflow. You will find a brief overview of the scripts with an explanation here: https://rshuhuachen.github.io/ms_load_grouse/</p>
</section>
<section id="data" class="level1">
<h1>Data</h1>
<section id="phenotypes" class="level2">
<h2 class="anchored" data-anchor-id="phenotypes">Phenotypes</h2>
<p>There are two files stored on GitHub containing phenotypes: one for lifetime phenotypic traits (including lifetime mating success) which can be found under <code>data/phenotypes/phenotypes_lifetime</code> both in .csv and .RData format. The second file is in long format, where each row represents a caught male at a lek, one data entry per male per year. This file contains the sexual and behavioural traits for each male, as well as annual mating success, and can be found under <code>data/phenotypes/phenotyes_annual</code> again in both .csv and .RData format.</p>
</section>
<section id="reference-genome" class="level2">
<h2 class="anchored" data-anchor-id="reference-genome">Reference genome</h2>
<p>The reference genome can be found on NCBI BioProject PRJNA1085187. Please download the file and store it in the folder <code>data/genomic/refgenome/</code> under name PO2979_Lyrurus_tetrix_black_grouse.RepeatMasked.fasta or edit the scripts that use the reference genome according to the updated file name.</p>
</section>
<section id="annotation" class="level2">
<h2 class="anchored" data-anchor-id="annotation">Annotation</h2>
<p>The genome annotation can be found in the folder <code>data/genomic/annotation/</code> under name PO2979_Lyrurus_tetrix_black_grouse.annotation.gff. The RNA seq data generated to annotate the genome can be found under NCBI Bio Accession <a href="https://www.ncbi.nlm.nih.gov/sra/SRX24353984%5Baccn%5D%5D">SRX24353984</a></p>
</section>
<section id="resequencing" class="level2">
<h2 class="anchored" data-anchor-id="resequencing">Resequencing</h2>
<p>The raw resequencing data can be found on NCBI BioProject <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA1085187">PRJNA1085187</a> under SRA study <a href="https://trace.ncbi.nlm.nih.gov/Traces/?view=study&amp;acc=SRP499251">SRP499251</a> with BioAccession Numbers SRR28526036 – SRR28526225. Please download the files and store it in the folder <code>data/genomic/raw/resequencing/</code>.</p>
</section>
<section id="hal-file" class="level2">
<h2 class="anchored" data-anchor-id="hal-file">HAL file</h2>
<p>The publicly available 363-avian multiple alignment file used for ancestral genome reconstruction and calculating GERP scores can be downloaded from https://cgl.gi.ucsc.edu/data/cactus/363-avian-2020.hal and should be stored under <code>data/genomic/intermediate/cactus</code>. Note that this file is a few hundred Gb’s in size!</p>
</section>
</section>
<section id="scripts" class="level1">
<h1>Scripts</h1>
<p>The scripts used for this study can be found in chronological order under <code>scripts/</code>. Note that a large portion of the analyses are computationally intensive and are recommended to be run on a cluster. Several scripts include headers for submitting through a job scheduler, such as SLURM/Sun Grid Engine which should be edited according to your local cluster/infrastructure.</p>
<p>A brief explanation of each step, divided into different folders under <code>scripts/</code>, can be found below. Additionally, you can find a README.md in some of the separate folders for a more thorough explanation of the more complex steps. Please note that pathnames should be edited according to your local device. R scripts should be opened from the .Rproj to allow easier work within the current directory, as the Rstudio Project will assume the root working directory as the default location. Note that some but not all processed files are stored in this repository (e.g.&nbsp;excluding .vcf files and GERP results) due to their large file sizes. However, all data used for plots are stored within this repository.</p>
<p>To allow easier reproducibility, you can use the conda environment listed in <code>src/envs/environment_load.yml</code>. Ensure you have conda installed on your local device, and then use <code>conda env create --name load --file src/envs/environment_load.yml</code> to install all the software and their correct versions as used in this manuscript. Ensure to activate the environment (<code>conda activate load</code>) before starting the analyses.</p>
<section id="brief-explanation-of-each-scripts-subdirectory" class="level2">
<h2 class="anchored" data-anchor-id="brief-explanation-of-each-scripts-subdirectory">Brief explanation of each script’s subdirectory</h2>
<p>1_variant_calling: here we align the sequences to the reference genome and call SNPs. We next filter the resulting .vcf file for quality as well as HWE, LD, MAF, etc.</p>
<p>2_cactus: here we manipulate the publicly available 363-avian multiple alignment file according to our needs (i.e.&nbsp;reducing the phylogenetic tree and adding focal genomes) which is a file required to determine the ancestral state of the genome and for GERP++ to infer evolutionary conservation</p>
<p>3_sex_chr: here we align the chicken Z chromosome to our genome to infer which scaffolds are most likely part of the black grouse sex chromosome</p>
<p>4_pop_structure: here we conduct a PCA to inspect population structure within our population</p>
<p>5_relatedness: here we assess relatedness coefficients based on genotypes in our population using NgsRelate and plink</p>
<p>6_snpeff_gerp: here we use two variant prioritisation tools to infer deleterious mutations: we run snpeff (first build a data base and then we polarise the genome to recode 1 as derived and 0 as ancestral) and gerp (first we run gerp across the genome and then select those that overlap with snps). Note the analyses should be performed with snpeff first, as the polarised genome is also used for the gerp - snp intersection to result in a vcf file with both snpeff annotations, gerp scores, and ancestral-derived recoded genotypes.</p>
<p>7_calculate_load: here we use the snpeff and gerp results to filter for the most deleterious mutations and calculated a proxy for total load, as well as homozygous and heterozygous load. Within this folder, we repeat the above per gene region (promoter, TSS, intron, exon) using the genome annotation.</p>
<p>8_models: here we use Bayesian mixed effect modelling to understand the relationship between mutation load and lifetime mating success (LMS), as well as their effects on sexual and behavioural traits. We also calculate allele frequencies in this directory</p>
<p>9_random_draws: in this subfolder, we use a subsampling approach to select a random set of mutations to control for the number of mutations that contribute to the mutation load estimate</p>
<p>10_plots: in this folder, we produce the plots used in the main manuscript as well as in the supplementary materials</p>
<p>Lastly, please see our full Methods for further details.</p>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>