<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>7&nbsp; Load per biological process â€“ Mutation Load in Black Grouse</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../qmd/7_random_sub.html" rel="next">
<link href="../qmd/5_per_region.html" rel="prev">
<link href="../plots/img/grouse_silhouette.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-7281d8b040e91ef2f2c9864091717679.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Mutation Load in Black Grouse</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../qmd/6_per_go.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Load per biological process</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Mutation load in black grouse</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/1_snpeff.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">SnpEff</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/2_gerp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">GERP</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/3_load.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Calculating mutation load</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/4_models.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Modelling fitness</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/5_per_region.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Load per genomic region</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/6_per_go.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Load per biological process</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/7_random_sub.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Random subsampling</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">7.1</span> Introduction</a>
  <ul class="collapse">
  <li><a href="#modelling" id="toc-modelling" class="nav-link" data-scroll-target="#modelling"><span class="header-section-number">7.1.1</span> Modelling</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">7.1.2</span> Results</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Load per biological process</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="introduction" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">7.1</span> Introduction</h2>
<p>Deleterious mutations that disrupt certain biological processes might affect LMS more negatively than in others. To test if there are certain biological processes where deleterious mutations have negative fitness effects, we hypothesized six processes and used gene ontology annotations. We used <a href="http://amigo.geneontology.org/amigo%5D">Amigo</a> to extract deleterious mutations associated with the GO terms: androgen metabolism, cellular respiration, developmental growth, immunity, muslce tissue development and response to oxidative stress. In the scripts below, youâ€™ll see how we calculate total load based on these GO terms.</p>
<section id="subset-mutations" class="level4" data-number="7.1.0.1">
<h4 data-number="7.1.0.1" class="anchored" data-anchor-id="subset-mutations"><span class="header-section-number">7.1.0.1</span> Subset mutations</h4>
<p>First, we subset the mutations located in genes associated with each GO term, and calculate total load based on that subset. We also divide it per genomic region (promoter, intron, exon). We work with lists to iterate the process with ease by looping over the list items.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">#### load packages ####</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(BiocManager, rtracklayer, GenomicFeatures, BiocGenerics, data.table, tidyverse, genomation, GenomicRanges, tibble)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="do">#### load the list of genes per GO term ####</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">"data/gene_lists_go"</span>, <span class="at">pattern =</span> <span class="st">"*.txt"</span>, <span class="at">full.names =</span> T)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(files)){</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">fread</span>(files[i])</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  list[[i]] <span class="ot">&lt;-</span> data</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(list) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"data/gene_lists_go/|.txt"</span>, <span class="st">""</span>, files)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="do">#### extract only the unique gene names per GO term ####</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  list[[i]] <span class="ot">&lt;-</span> list[[i]] <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">`</span><span class="at">V1</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">go_term =</span> <span class="fu">names</span>(list[i])) <span class="sc">%&gt;%</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene_id =</span> <span class="st">`</span><span class="at">V1</span><span class="st">`</span>) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">gene_id =</span> <span class="fu">toupper</span>(gene_id))</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="do">#### GERP ####</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="do">#### import gerp data ####</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"output/load/gerp/gerp_over4.RData"</span>)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>gerp<span class="sc">$</span>chr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">";"</span>, <span class="st">"__"</span>, gerp<span class="sc">$</span>chr)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>gerp<span class="sc">$</span>chr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"="</span>, <span class="st">"_"</span>, gerp<span class="sc">$</span>chr)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>gerp<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">paste0</span>(gerp<span class="sc">$</span>chr, <span class="st">"_"</span>, gerp<span class="sc">$</span>pos)</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="co"># extract only the positions for annotation</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>gerp_snps <span class="ot">&lt;-</span> gerp <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(chr, pos)</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>gerp_snps<span class="sc">$</span>end <span class="ot">&lt;-</span> gerp_snps<span class="sc">$</span>pos</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>gerp_snps<span class="sc">$</span>start <span class="ot">&lt;-</span> gerp_snps<span class="sc">$</span>pos</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>gerp_gr <span class="ot">&lt;-</span> <span class="fu">as</span>(gerp_snps, <span class="st">"GRanges"</span>)</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="do">#### import gene regions ####</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="do">### load annotation data ####</span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>annotation_dir <span class="ot">&lt;-</span> <span class="st">"data/genomic/annotation"</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>promoter<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/promoters.gff3"</span>)))</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>exons_gene<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/exons_gene.gff3"</span>)))</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>introns<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/introns_transcripts.gff3"</span>)))</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="do">#### annotate gerp mutations ####</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>gerp_promoter <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">mergeByOverlaps</span>(gerp_gr, promoter)) <span class="sc">%&gt;%</span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region_promoter"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">gerp_gr.seqnames</span><span class="st">`</span>, pos, region_promoter,ID)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">chr =</span> gerp_gr.seqnames)</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>gerp_exons <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">mergeByOverlaps</span>(gerp_gr, exons_gene))<span class="sc">%&gt;%</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region_exon"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(gerp_gr.seqnames, pos, region_exon,ID)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">chr =</span> gerp_gr.seqnames)</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>gerp_intron <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">mergeByOverlaps</span>(gerp_gr, introns))<span class="sc">%&gt;%</span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region_intron"</span> <span class="ot">=</span> <span class="st">"1"</span>) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(gerp_gr.seqnames, pos, region_intron,ID)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">chr =</span> gerp_gr.seqnames)</span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>gerp_list <span class="ot">&lt;-</span> <span class="fu">list</span>(gerp_promoter, gerp_exons, gerp_intron)</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="do">#### merge with 'similar to' column to get gene IDs ####</span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>lookup <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"../grouse-annotation/data/lookup_ANN_gene_id.txt"</span>)</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(lookup) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ann"</span>, <span class="st">"gene_id"</span>)</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gerp_list)){</span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>  gerp_list[[i]] <span class="ot">&lt;-</span> <span class="fu">left_join</span>(gerp_list[[i]], lookup, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"ann"</span>))</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(gerp_list[[i]])[<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ann"</span>)</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>  gerp_list[[i]]<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> <span class="fu">toupper</span>(gerp_list[[i]]<span class="sc">$</span>gene_id)</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a><span class="do">#### extract snps per GO term ####</span></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>go_snps_promo <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>  overlap <span class="ot">&lt;-</span> <span class="fu">subset</span>(gerp_list[[<span class="dv">1</span>]], gene_id <span class="sc">%in%</span> list[[i]]<span class="sc">$</span>gene_id)</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a>  go_snps_promo[[i]] <span class="ot">&lt;-</span> overlap}</span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>go_snps_exon <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a>  overlap <span class="ot">&lt;-</span> <span class="fu">subset</span>(gerp_list[[<span class="dv">2</span>]], gene_id <span class="sc">%in%</span> list[[i]]<span class="sc">$</span>gene_id)</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>  go_snps_exon[[i]] <span class="ot">&lt;-</span> overlap}</span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a>go_snps_intron <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>  overlap <span class="ot">&lt;-</span> <span class="fu">subset</span>(gerp_list[[<span class="dv">3</span>]], gene_id <span class="sc">%in%</span> list[[i]]<span class="sc">$</span>gene_id)</span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>  go_snps_intron[[i]] <span class="ot">&lt;-</span> overlap}</span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="co"># name</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(go_snps_promo) <span class="ot">&lt;-</span> <span class="fu">names</span>(list)</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(go_snps_exon) <span class="ot">&lt;-</span> <span class="fu">names</span>(list)</span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(go_snps_intron) <span class="ot">&lt;-</span> <span class="fu">names</span>(list)</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>summary_snps_per_go <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>  snps_go <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">go =</span> <span class="fu">names</span>(list[i]),</span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>                        <span class="at">snps_promo =</span> <span class="fu">nrow</span>(<span class="fu">subset</span>(go_snps_promo[[i]], region_promoter <span class="sc">==</span> <span class="dv">1</span>)),</span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a>                        <span class="at">snps_exon =</span> <span class="fu">nrow</span>(<span class="fu">subset</span>(go_snps_exon[[i]], region_exon <span class="sc">==</span> <span class="dv">1</span>)),</span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>                        <span class="at">snps_intron =</span> <span class="fu">nrow</span>(<span class="fu">subset</span>(go_snps_intron[[i]], region_intron <span class="sc">==</span> <span class="dv">1</span>)),</span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a>                        <span class="at">og_genes_total =</span> <span class="fu">nrow</span>(list[[i]]))</span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a>  summary_snps_per_go <span class="ot">&lt;-</span> <span class="fu">rbind</span>(summary_snps_per_go, snps_go)</span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a><span class="co"># how many genes</span></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>genes_n <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>  promo <span class="ot">&lt;-</span> <span class="fu">subset</span>(go_snps_promo[[i]], region_promoter <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a>  exon <span class="ot">&lt;-</span> <span class="fu">subset</span>(go_snps_exon[[i]], region_exon <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>  intron <span class="ot">&lt;-</span> <span class="fu">subset</span>(go_snps_intron[[i]], region_intron <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(promo<span class="sc">$</span>gene_id, exon<span class="sc">$</span>gene_id, intron<span class="sc">$</span>gene_id))</span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>  genes_go <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">go =</span> <span class="fu">names</span>(list[i]),</span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>                        <span class="at">n_genes =</span> <span class="fu">length</span>(genes))</span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>  genes_n <span class="ot">&lt;-</span> <span class="fu">rbind</span>(genes_n, genes_go)</span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(genes_n, <span class="at">file =</span> <span class="st">"output/biological_pathways/n_genes_gerp_per_go.tsv"</span>)</span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a><span class="co">#note: there are snps for every go term!</span></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a><span class="do">#### extract genotypes per GO term ####</span></span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a><span class="co">#promo</span></span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a>genotypes_go_promo <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(go_snps_promo)){</span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a>  go_snps_promo[[i]]<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">paste0</span>(go_snps_promo[[i]]<span class="sc">$</span>chr, <span class="st">"_"</span>, go_snps_promo[[i]]<span class="sc">$</span>pos)</span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a>  geno <span class="ot">&lt;-</span> <span class="fu">subset</span>(gerp, chr_pos <span class="sc">%in%</span> go_snps_promo[[i]]<span class="sc">$</span>chr_pos)</span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a>  geno<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a>  genotypes_go_promo[[i]] <span class="ot">&lt;-</span> geno</span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(genotypes_go_promo) <span class="ot">&lt;-</span> <span class="fu">names</span>(list)</span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a><span class="co">#exon</span></span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a>genotypes_go_exon <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(go_snps_exon)){</span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a>  go_snps_exon[[i]]<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">paste0</span>(go_snps_exon[[i]]<span class="sc">$</span>chr, <span class="st">"_"</span>, go_snps_exon[[i]]<span class="sc">$</span>pos)</span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a>  geno <span class="ot">&lt;-</span> <span class="fu">subset</span>(gerp, chr_pos <span class="sc">%in%</span> go_snps_exon[[i]]<span class="sc">$</span>chr_pos)</span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a>  geno<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a>  genotypes_go_exon[[i]] <span class="ot">&lt;-</span> geno</span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(genotypes_go_exon) <span class="ot">&lt;-</span> <span class="fu">names</span>(list)</span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a><span class="co">#intron</span></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a>genotypes_go_intron <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(go_snps_intron)){</span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a>  go_snps_intron[[i]]<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">paste0</span>(go_snps_intron[[i]]<span class="sc">$</span>chr, <span class="st">"_"</span>, go_snps_intron[[i]]<span class="sc">$</span>pos)</span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a>  geno <span class="ot">&lt;-</span> <span class="fu">subset</span>(gerp, chr_pos <span class="sc">%in%</span> go_snps_intron[[i]]<span class="sc">$</span>chr_pos)</span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a>  geno<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a>  genotypes_go_intron[[i]] <span class="ot">&lt;-</span> geno</span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(genotypes_go_intron) <span class="ot">&lt;-</span> <span class="fu">names</span>(list)</span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a><span class="do">#### SnpEff ####</span></span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a><span class="do">#### extract only the unique gene names per GO term ####</span></span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a><span class="do">#### import snpeff data ####</span></span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"output/load/snpeff/snpeff_high.RData"</span>)</span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a>snpeff<span class="sc">$</span>CHROM <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">";"</span>, <span class="st">"__"</span>, snpeff<span class="sc">$</span>CHROM)</span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a>snpeff<span class="sc">$</span>CHROM <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"="</span>, <span class="st">"_"</span>, snpeff<span class="sc">$</span>CHROM)</span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a>snpeff<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">paste0</span>(snpeff<span class="sc">$</span>CHROM, <span class="st">"_"</span>, snpeff<span class="sc">$</span>POS)</span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a><span class="co"># extract only the positions for annotation</span></span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a>snpeff_snps <span class="ot">&lt;-</span> snpeff <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(CHROM, POS)</span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a>snpeff_snps<span class="sc">$</span>end <span class="ot">&lt;-</span> snpeff_snps<span class="sc">$</span>POS</span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a>snpeff_snps<span class="sc">$</span>start <span class="ot">&lt;-</span> snpeff_snps<span class="sc">$</span>POS</span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a>snpeff_gr <span class="ot">&lt;-</span> <span class="fu">as</span>(snpeff_snps, <span class="st">"GRanges"</span>)</span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a><span class="do">#### import gene regions ####</span></span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a><span class="do">#### annotate snpeff mutations ####</span></span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a>snpeff_promoter <span class="ot">&lt;-</span> <span class="fu">mergeByOverlaps</span>(snpeff_gr, promoter)</span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a>snpeff_promoter <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(snpeff_promoter<span class="sc">@</span>listData)</span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a>snpeff_promoter <span class="ot">&lt;-</span> snpeff_promoter <span class="sc">%&gt;%</span></span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region_promoter"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">`</span><span class="at">snpeff_gr.seqnames</span><span class="st">`</span>, POS, region_promoter,ID)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">chr =</span> snpeff_gr.seqnames)</span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a>snpeff_exons <span class="ot">&lt;-</span> <span class="fu">mergeByOverlaps</span>(snpeff_gr, exons_gene)</span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a>snpeff_exons <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(snpeff_exons<span class="sc">@</span>listData)</span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a>snpeff_exons <span class="ot">&lt;-</span> snpeff_exons<span class="sc">%&gt;%</span></span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region_exon"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(snpeff_gr.seqnames, POS, region_exon,ID)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">chr =</span> snpeff_gr.seqnames)</span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a>snpeff_introns <span class="ot">&lt;-</span> <span class="fu">mergeByOverlaps</span>(snpeff_gr, introns)</span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a>snpeff_introns <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(snpeff_introns<span class="sc">@</span>listData)</span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a>snpeff_introns <span class="ot">&lt;-</span> snpeff_introns<span class="sc">%&gt;%</span></span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region_intron"</span> <span class="ot">=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">unique</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(snpeff_gr.seqnames, POS, region_intron,ID)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">chr =</span> snpeff_gr.seqnames)</span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a>snpeff_list <span class="ot">&lt;-</span> <span class="fu">list</span>(snpeff_promoter, snpeff_exons, snpeff_introns)</span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a><span class="do">#### merge with 'similar to' column to get gene IDs ####</span></span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a>lookup <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"data/genomic/annotation/lookup_ANN_gene_id.txt"</span>)</span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(lookup) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ann"</span>, <span class="st">"gene_id"</span>)</span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(snpeff_list)){</span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a>  snpeff_list[[i]] <span class="ot">&lt;-</span> <span class="fu">left_join</span>(snpeff_list[[i]], lookup, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"ID"</span> <span class="ot">=</span> <span class="st">"ann"</span>))</span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(snpeff_list[[i]])[<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ann"</span>)</span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a>  snpeff_list[[i]]<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> <span class="fu">toupper</span>(snpeff_list[[i]]<span class="sc">$</span>gene_id)</span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a><span class="do">#### extract snps per GO term ####</span></span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a>go_snps_promo_snpeff <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-199"><a href="#cb1-199" aria-hidden="true" tabindex="-1"></a>  overlap <span class="ot">&lt;-</span> <span class="fu">subset</span>(snpeff_list[[<span class="dv">1</span>]], gene_id <span class="sc">%in%</span> list[[i]]<span class="sc">$</span>gene_id)</span>
<span id="cb1-200"><a href="#cb1-200" aria-hidden="true" tabindex="-1"></a>  go_snps_promo_snpeff[[i]] <span class="ot">&lt;-</span> overlap}</span>
<span id="cb1-201"><a href="#cb1-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-202"><a href="#cb1-202" aria-hidden="true" tabindex="-1"></a>go_snps_exon_snpeff <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-203"><a href="#cb1-203" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-204"><a href="#cb1-204" aria-hidden="true" tabindex="-1"></a>  overlap <span class="ot">&lt;-</span> <span class="fu">subset</span>(snpeff_list[[<span class="dv">2</span>]], gene_id <span class="sc">%in%</span> list[[i]]<span class="sc">$</span>gene_id)</span>
<span id="cb1-205"><a href="#cb1-205" aria-hidden="true" tabindex="-1"></a>  go_snps_exon_snpeff[[i]] <span class="ot">&lt;-</span> overlap}</span>
<span id="cb1-206"><a href="#cb1-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-207"><a href="#cb1-207" aria-hidden="true" tabindex="-1"></a>go_snps_intron_snpeff <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-208"><a href="#cb1-208" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-209"><a href="#cb1-209" aria-hidden="true" tabindex="-1"></a>  overlap <span class="ot">&lt;-</span> <span class="fu">subset</span>(snpeff_list[[<span class="dv">3</span>]], gene_id <span class="sc">%in%</span> list[[i]]<span class="sc">$</span>gene_id)</span>
<span id="cb1-210"><a href="#cb1-210" aria-hidden="true" tabindex="-1"></a>  go_snps_intron_snpeff[[i]] <span class="ot">&lt;-</span> overlap}</span>
<span id="cb1-211"><a href="#cb1-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-212"><a href="#cb1-212" aria-hidden="true" tabindex="-1"></a><span class="co"># name</span></span>
<span id="cb1-213"><a href="#cb1-213" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(go_snps_promo_snpeff) <span class="ot">&lt;-</span> <span class="fu">names</span>(list)</span>
<span id="cb1-214"><a href="#cb1-214" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(go_snps_exon_snpeff) <span class="ot">&lt;-</span> <span class="fu">names</span>(list)</span>
<span id="cb1-215"><a href="#cb1-215" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(go_snps_intron_snpeff) <span class="ot">&lt;-</span> <span class="fu">names</span>(list)</span>
<span id="cb1-216"><a href="#cb1-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-217"><a href="#cb1-217" aria-hidden="true" tabindex="-1"></a>summary_snps_per_go <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb1-218"><a href="#cb1-218" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-219"><a href="#cb1-219" aria-hidden="true" tabindex="-1"></a>  snps_go <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">go =</span> <span class="fu">names</span>(list[i]),</span>
<span id="cb1-220"><a href="#cb1-220" aria-hidden="true" tabindex="-1"></a>                        <span class="at">snps_promo =</span> <span class="fu">nrow</span>(<span class="fu">subset</span>(go_snps_promo_snpeff[[i]], region_promoter <span class="sc">==</span> <span class="dv">1</span>)),</span>
<span id="cb1-221"><a href="#cb1-221" aria-hidden="true" tabindex="-1"></a>                        <span class="at">snps_exon =</span> <span class="fu">nrow</span>(<span class="fu">subset</span>(go_snps_exon_snpeff[[i]], region_exon <span class="sc">==</span> <span class="dv">1</span>)),</span>
<span id="cb1-222"><a href="#cb1-222" aria-hidden="true" tabindex="-1"></a>                        <span class="at">snps_intron =</span> <span class="fu">nrow</span>(<span class="fu">subset</span>(go_snps_intron_snpeff[[i]], region_intron <span class="sc">==</span> <span class="dv">1</span>)),</span>
<span id="cb1-223"><a href="#cb1-223" aria-hidden="true" tabindex="-1"></a>                        <span class="at">og_genes_total =</span> <span class="fu">nrow</span>(list[[i]]))</span>
<span id="cb1-224"><a href="#cb1-224" aria-hidden="true" tabindex="-1"></a>  summary_snps_per_go <span class="ot">&lt;-</span> <span class="fu">rbind</span>(summary_snps_per_go, snps_go)</span>
<span id="cb1-225"><a href="#cb1-225" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-226"><a href="#cb1-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-227"><a href="#cb1-227" aria-hidden="true" tabindex="-1"></a>genes_n_snpeff <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb1-228"><a href="#cb1-228" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb1-229"><a href="#cb1-229" aria-hidden="true" tabindex="-1"></a>  promo <span class="ot">&lt;-</span> <span class="fu">subset</span>(go_snps_promo_snpeff[[i]], region_promoter <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb1-230"><a href="#cb1-230" aria-hidden="true" tabindex="-1"></a>  exon <span class="ot">&lt;-</span> <span class="fu">subset</span>(go_snps_exon_snpeff[[i]], region_exon <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb1-231"><a href="#cb1-231" aria-hidden="true" tabindex="-1"></a>  intron <span class="ot">&lt;-</span> <span class="fu">subset</span>(go_snps_intron_snpeff[[i]], region_intron <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb1-232"><a href="#cb1-232" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(promo<span class="sc">$</span>gene_id, exon<span class="sc">$</span>gene_id, intron<span class="sc">$</span>gene_id))</span>
<span id="cb1-233"><a href="#cb1-233" aria-hidden="true" tabindex="-1"></a>  genes_go <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">go =</span> <span class="fu">names</span>(list[i]),</span>
<span id="cb1-234"><a href="#cb1-234" aria-hidden="true" tabindex="-1"></a>                         <span class="at">n_genes =</span> <span class="fu">length</span>(genes))</span>
<span id="cb1-235"><a href="#cb1-235" aria-hidden="true" tabindex="-1"></a>  genes_n_snpeff <span class="ot">&lt;-</span> <span class="fu">rbind</span>(genes_n_snpeff, genes_go)</span>
<span id="cb1-236"><a href="#cb1-236" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-237"><a href="#cb1-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-238"><a href="#cb1-238" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(genes_n_snpeff, <span class="at">file =</span> <span class="st">"output/biological_pathways/n_genes_snpeff_per_go.tsv"</span>)</span>
<span id="cb1-239"><a href="#cb1-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-240"><a href="#cb1-240" aria-hidden="true" tabindex="-1"></a><span class="co">#note: there are snps for every go term!</span></span>
<span id="cb1-241"><a href="#cb1-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-242"><a href="#cb1-242" aria-hidden="true" tabindex="-1"></a><span class="do">#### extract genotypes per GO term ####</span></span>
<span id="cb1-243"><a href="#cb1-243" aria-hidden="true" tabindex="-1"></a><span class="co">#only select those with &gt;10 snps</span></span>
<span id="cb1-244"><a href="#cb1-244" aria-hidden="true" tabindex="-1"></a>go_snps_promo_snpeff <span class="ot">&lt;-</span> go_snps_promo_snpeff[<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span><span class="sc">:</span><span class="dv">9</span>)]</span>
<span id="cb1-245"><a href="#cb1-245" aria-hidden="true" tabindex="-1"></a>go_snps_exon_snpeff <span class="ot">&lt;-</span> go_snps_exon_snpeff[<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span><span class="sc">:</span><span class="dv">9</span>)]</span>
<span id="cb1-246"><a href="#cb1-246" aria-hidden="true" tabindex="-1"></a>go_snps_intron_snpeff <span class="ot">&lt;-</span> go_snps_intron_snpeff[<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span><span class="sc">:</span><span class="dv">9</span>)]</span>
<span id="cb1-247"><a href="#cb1-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-248"><a href="#cb1-248" aria-hidden="true" tabindex="-1"></a><span class="co">#promo</span></span>
<span id="cb1-249"><a href="#cb1-249" aria-hidden="true" tabindex="-1"></a>genotypes_go_promo_snpeff <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-250"><a href="#cb1-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-251"><a href="#cb1-251" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(go_snps_promo_snpeff)){</span>
<span id="cb1-252"><a href="#cb1-252" aria-hidden="true" tabindex="-1"></a>  go_snps_promo_snpeff[[i]]<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">paste0</span>(go_snps_promo_snpeff[[i]]<span class="sc">$</span>chr, <span class="st">"_"</span>, go_snps_promo_snpeff[[i]]<span class="sc">$</span>POS)</span>
<span id="cb1-253"><a href="#cb1-253" aria-hidden="true" tabindex="-1"></a>  geno <span class="ot">&lt;-</span> <span class="fu">subset</span>(snpeff, chr_pos <span class="sc">%in%</span> go_snps_promo_snpeff[[i]]<span class="sc">$</span>chr_pos)</span>
<span id="cb1-254"><a href="#cb1-254" aria-hidden="true" tabindex="-1"></a>  geno<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-255"><a href="#cb1-255" aria-hidden="true" tabindex="-1"></a>  genotypes_go_promo_snpeff[[i]] <span class="ot">&lt;-</span> geno</span>
<span id="cb1-256"><a href="#cb1-256" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-257"><a href="#cb1-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-258"><a href="#cb1-258" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(genotypes_go_promo_snpeff) <span class="ot">&lt;-</span> <span class="fu">names</span>(go_snps_promo_snpeff)</span>
<span id="cb1-259"><a href="#cb1-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-260"><a href="#cb1-260" aria-hidden="true" tabindex="-1"></a><span class="co">#exon</span></span>
<span id="cb1-261"><a href="#cb1-261" aria-hidden="true" tabindex="-1"></a>genotypes_go_exon_snpeff <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-262"><a href="#cb1-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-263"><a href="#cb1-263" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(go_snps_exon_snpeff)){</span>
<span id="cb1-264"><a href="#cb1-264" aria-hidden="true" tabindex="-1"></a>  go_snps_exon_snpeff[[i]]<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">paste0</span>(go_snps_exon_snpeff[[i]]<span class="sc">$</span>chr, <span class="st">"_"</span>, go_snps_exon_snpeff[[i]]<span class="sc">$</span>POS)</span>
<span id="cb1-265"><a href="#cb1-265" aria-hidden="true" tabindex="-1"></a>  geno <span class="ot">&lt;-</span> <span class="fu">subset</span>(snpeff, chr_pos <span class="sc">%in%</span> go_snps_exon_snpeff[[i]]<span class="sc">$</span>chr_pos)</span>
<span id="cb1-266"><a href="#cb1-266" aria-hidden="true" tabindex="-1"></a>  geno<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-267"><a href="#cb1-267" aria-hidden="true" tabindex="-1"></a>  genotypes_go_exon_snpeff[[i]] <span class="ot">&lt;-</span> geno</span>
<span id="cb1-268"><a href="#cb1-268" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-269"><a href="#cb1-269" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(genotypes_go_exon_snpeff) <span class="ot">&lt;-</span> <span class="fu">names</span>(go_snps_exon_snpeff)</span>
<span id="cb1-270"><a href="#cb1-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-271"><a href="#cb1-271" aria-hidden="true" tabindex="-1"></a><span class="co">#intron</span></span>
<span id="cb1-272"><a href="#cb1-272" aria-hidden="true" tabindex="-1"></a>genotypes_go_intron_snpeff <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-273"><a href="#cb1-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-274"><a href="#cb1-274" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(go_snps_intron_snpeff)){</span>
<span id="cb1-275"><a href="#cb1-275" aria-hidden="true" tabindex="-1"></a>  go_snps_intron_snpeff[[i]]<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">paste0</span>(go_snps_intron_snpeff[[i]]<span class="sc">$</span>chr, <span class="st">"_"</span>, go_snps_intron_snpeff[[i]]<span class="sc">$</span>POS)</span>
<span id="cb1-276"><a href="#cb1-276" aria-hidden="true" tabindex="-1"></a>  geno <span class="ot">&lt;-</span> <span class="fu">subset</span>(snpeff, chr_pos <span class="sc">%in%</span> go_snps_intron_snpeff[[i]]<span class="sc">$</span>chr_pos)</span>
<span id="cb1-277"><a href="#cb1-277" aria-hidden="true" tabindex="-1"></a>  geno<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-278"><a href="#cb1-278" aria-hidden="true" tabindex="-1"></a>  genotypes_go_intron_snpeff[[i]] <span class="ot">&lt;-</span> geno</span>
<span id="cb1-279"><a href="#cb1-279" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-280"><a href="#cb1-280" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(genotypes_go_intron_snpeff) <span class="ot">&lt;-</span> <span class="fu">names</span>(go_snps_intron_snpeff)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="calculate-mutation-load" class="level4" data-number="7.1.0.2">
<h4 data-number="7.1.0.2" class="anchored" data-anchor-id="calculate-mutation-load"><span class="header-section-number">7.1.0.2</span> Calculate mutation load</h4>
<p>Then, we calculate the total load based on these subsets, again for GERP and SnpEff separately.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">#### calculate mutation load based on mutations in each GO term ####</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="do">### GERP</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"scripts/7_calculate_load/function_calculate_load.R"</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>loads <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list)){</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#promo</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  load_promo <span class="ot">&lt;-</span> <span class="fu">calculate_load_gerp</span>(<span class="fu">unique</span>(genotypes_go_promo[[i]]), <span class="at">loadtype =</span> <span class="fu">names</span>(genotypes_go_promo[i]), <span class="at">output_vcf =</span> F)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  load_promo<span class="sc">$</span>method <span class="ot">&lt;-</span> <span class="st">"gerp"</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  load_promo<span class="sc">$</span>region <span class="ot">=</span> <span class="st">"promo"</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  loads <span class="ot">&lt;-</span> <span class="fu">rbind</span>(loads, load_promo)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#exon</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  load_exon <span class="ot">&lt;-</span> <span class="fu">calculate_load_gerp</span>(genotypes_go_exon[[i]], <span class="at">loadtype =</span> <span class="fu">names</span>(genotypes_go_exon[i]), <span class="at">output_vcf =</span> F)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  load_exon<span class="sc">$</span>method <span class="ot">&lt;-</span> <span class="st">"gerp"</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  load_exon<span class="sc">$</span>region <span class="ot">=</span> <span class="st">"exon"</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  loads <span class="ot">&lt;-</span> <span class="fu">rbind</span>(loads, load_exon)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="co">#intron</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>  load_intron <span class="ot">&lt;-</span> <span class="fu">calculate_load_gerp</span>(genotypes_go_intron[[i]], <span class="at">loadtype =</span> <span class="fu">names</span>(genotypes_go_intron[i]), <span class="at">output_vcf =</span> F)</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  load_intron<span class="sc">$</span>method <span class="ot">&lt;-</span> <span class="st">"gerp"</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  load_intron<span class="sc">$</span>region <span class="ot">=</span> <span class="st">"intron"</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  loads <span class="ot">&lt;-</span> <span class="fu">rbind</span>(loads, load_intron)</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a><span class="do">#### SnpEff</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>loads_snpeff <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(genotypes_go_promo_snpeff)){</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>  <span class="co">#promo</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>  load <span class="ot">&lt;-</span> <span class="fu">calculate_load_snpeff</span>(genotypes_go_promo_snpeff[[i]], <span class="at">loadtype =</span> <span class="fu">names</span>(genotypes_go_promo_snpeff[i]), <span class="at">output_vcf =</span> F)</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>  load<span class="sc">$</span>method <span class="ot">&lt;-</span> <span class="st">"snpeff"</span></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>  load<span class="sc">$</span>region <span class="ot">=</span> <span class="st">"promo"</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  loads_snpeff <span class="ot">&lt;-</span> <span class="fu">rbind</span>(loads_snpeff, load)</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  <span class="co">#exon</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  load <span class="ot">&lt;-</span> <span class="fu">calculate_load_snpeff</span>(genotypes_go_exon_snpeff[[i]], <span class="at">loadtype =</span> <span class="fu">names</span>(genotypes_go_exon_snpeff[i]), <span class="at">output_vcf =</span> F)</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>  load<span class="sc">$</span>method <span class="ot">&lt;-</span> <span class="st">"snpeff"</span></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>  load<span class="sc">$</span>region <span class="ot">=</span> <span class="st">"exon"</span></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>  loads_snpeff <span class="ot">&lt;-</span> <span class="fu">rbind</span>(loads_snpeff, load)</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>  <span class="co">#intron</span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>  load <span class="ot">&lt;-</span> <span class="fu">calculate_load_snpeff</span>(genotypes_go_intron_snpeff[[i]], <span class="at">loadtype =</span> <span class="fu">names</span>(genotypes_go_intron_snpeff[i]), <span class="at">output_vcf =</span> F)</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>  load<span class="sc">$</span>method <span class="ot">&lt;-</span> <span class="st">"snpeff"</span></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>  load<span class="sc">$</span>region <span class="ot">=</span> <span class="st">"intron"</span></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>  loads_snpeff <span class="ot">&lt;-</span> <span class="fu">rbind</span>(loads_snpeff, load)</span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>loads <span class="ot">&lt;-</span> <span class="fu">rbind</span>(loads, loads_snpeff)</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(loads, <span class="at">file =</span> <span class="st">"output/biological_pathways/loads_per_go_term_per_region.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="modelling" class="level3" data-number="7.1.1">
<h3 data-number="7.1.1" class="anchored" data-anchor-id="modelling"><span class="header-section-number">7.1.1</span> Modelling</h3>
<p>We then computed the exact same models as the total load models presented before with the following model structure:</p>
<p><code>LMS ~ scale(total_load) + core + (1|site)</code></p>
<p>Where total load is not a measure taken from a subset of deleterious mutations located within a specific genomic region and associated with a certain subset of genes.</p>
</section>
<section id="results" class="level3" data-number="7.1.2">
<h3 data-number="7.1.2" class="anchored" data-anchor-id="results"><span class="header-section-number">7.1.2</span> Results</h3>
<p>Below you find the posterior distributions:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../plots/extended/extended_5_go_gerp.png" class="img-fluid figure-img"></p>
<figcaption>Results per genomic region per GO term for GERP</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../plots/extended/extended_6_go_snpeff.png" class="img-fluid figure-img"></p>
<figcaption>Results per genomic region per GO term for SnpEff</figcaption>
</figure>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table); <span class="fu">library</span>(kableExtra)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>gerp <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"../output/biological_pathways/intervals_gerp.tsv"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>gerp <span class="sc">%&gt;%</span> <span class="fu">kbl</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">parameter</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">region</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">go</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">median</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">ci_95</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">ci_80</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Androgen metabolism</td>
<td style="text-align: right;">-0.17</td>
<td style="text-align: left;">-0.25, -0.1</td>
<td style="text-align: left;">-0.23, -0.12</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Cellular respiration</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: left;">-0.06, 0.09</td>
<td style="text-align: left;">-0.03, 0.07</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Developmental growth</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: left;">-0.04, 0.13</td>
<td style="text-align: left;">-0.01, 0.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Immune response</td>
<td style="text-align: right;">0.14</td>
<td style="text-align: left;">0.06, 0.23</td>
<td style="text-align: left;">0.08, 0.2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Muscle tissue development</td>
<td style="text-align: right;">-0.02</td>
<td style="text-align: left;">-0.1, 0.07</td>
<td style="text-align: left;">-0.07, 0.04</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Response to oxidative stress</td>
<td style="text-align: right;">-0.17</td>
<td style="text-align: left;">-0.25, -0.09</td>
<td style="text-align: left;">-0.22, -0.11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Androgen metabolism</td>
<td style="text-align: right;">0.41</td>
<td style="text-align: left;">0.32, 0.49</td>
<td style="text-align: left;">0.35, 0.46</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Cellular respiration</td>
<td style="text-align: right;">0.33</td>
<td style="text-align: left;">0.23, 0.43</td>
<td style="text-align: left;">0.27, 0.4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Developmental growth</td>
<td style="text-align: right;">0.09</td>
<td style="text-align: left;">0, 0.18</td>
<td style="text-align: left;">0.03, 0.15</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Immune response</td>
<td style="text-align: right;">-0.29</td>
<td style="text-align: left;">-0.38, -0.21</td>
<td style="text-align: left;">-0.35, -0.24</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Muscle tissue development</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: left;">0.01, 0.15</td>
<td style="text-align: left;">0.03, 0.13</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Response to oxidative stress</td>
<td style="text-align: right;">-0.03</td>
<td style="text-align: left;">-0.12, 0.06</td>
<td style="text-align: left;">-0.09, 0.03</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Androgen metabolism</td>
<td style="text-align: right;">-0.33</td>
<td style="text-align: left;">-0.4, -0.25</td>
<td style="text-align: left;">-0.38, -0.28</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Cellular respiration</td>
<td style="text-align: right;">-0.25</td>
<td style="text-align: left;">-0.34, -0.16</td>
<td style="text-align: left;">-0.3, -0.19</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Developmental growth</td>
<td style="text-align: right;">-0.22</td>
<td style="text-align: left;">-0.3, -0.15</td>
<td style="text-align: left;">-0.27, -0.17</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Immune response</td>
<td style="text-align: right;">-0.23</td>
<td style="text-align: left;">-0.3, -0.15</td>
<td style="text-align: left;">-0.28, -0.18</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Muscle tissue development</td>
<td style="text-align: right;">-0.09</td>
<td style="text-align: left;">-0.17, -0.01</td>
<td style="text-align: left;">-0.14, -0.04</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Response to oxidative stress</td>
<td style="text-align: right;">-0.11</td>
<td style="text-align: left;">-0.19, -0.02</td>
<td style="text-align: left;">-0.16, -0.05</td>
</tr>
</tbody>
</table>


</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>high <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"../output/biological_pathways/intervals_high.tsv"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>high <span class="sc">%&gt;%</span> <span class="fu">kbl</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">parameter</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">region</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">go</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">median</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">ci_95</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">ci_80</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Cellular respiration</td>
<td style="text-align: right;">0.19</td>
<td style="text-align: left;">0.1, 0.28</td>
<td style="text-align: left;">0.13, 0.25</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Developmental growth</td>
<td style="text-align: right;">-0.02</td>
<td style="text-align: left;">-0.1, 0.06</td>
<td style="text-align: left;">-0.07, 0.03</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Immune response</td>
<td style="text-align: right;">-0.14</td>
<td style="text-align: left;">-0.23, -0.05</td>
<td style="text-align: left;">-0.2, -0.08</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Muscle tissue development</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: left;">-0.07, 0.08</td>
<td style="text-align: left;">-0.04, 0.06</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Response to oxidative stress</td>
<td style="text-align: right;">-0.07</td>
<td style="text-align: left;">-0.15, 0.02</td>
<td style="text-align: left;">-0.12, -0.02</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Cellular respiration</td>
<td style="text-align: right;">0.09</td>
<td style="text-align: left;">0.02, 0.17</td>
<td style="text-align: left;">0.04, 0.14</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Developmental growth</td>
<td style="text-align: right;">0.29</td>
<td style="text-align: left;">0.2, 0.38</td>
<td style="text-align: left;">0.23, 0.35</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Immune response</td>
<td style="text-align: right;">-0.08</td>
<td style="text-align: left;">-0.15, 0</td>
<td style="text-align: left;">-0.13, -0.03</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Muscle tissue development</td>
<td style="text-align: right;">0.20</td>
<td style="text-align: left;">0.11, 0.29</td>
<td style="text-align: left;">0.14, 0.26</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Response to oxidative stress</td>
<td style="text-align: right;">0.37</td>
<td style="text-align: left;">0.28, 0.45</td>
<td style="text-align: left;">0.31, 0.42</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Cellular respiration</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: left;">-0.09, 0.09</td>
<td style="text-align: left;">-0.06, 0.06</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Developmental growth</td>
<td style="text-align: right;">-0.05</td>
<td style="text-align: left;">-0.14, 0.03</td>
<td style="text-align: left;">-0.11, 0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Immune response</td>
<td style="text-align: right;">-0.25</td>
<td style="text-align: left;">-0.34, -0.17</td>
<td style="text-align: left;">-0.31, -0.2</td>
</tr>
<tr class="even">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Muscle tissue development</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: left;">-0.03, 0.13</td>
<td style="text-align: left;">0, 0.1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">b_scaletotal_load</td>
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Response to oxidative stress</td>
<td style="text-align: right;">0.11</td>
<td style="text-align: left;">0.03, 0.19</td>
<td style="text-align: left;">0.06, 0.16</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>These results indicate the results really depend on the GO term and region in question. But three patterns are clear: - GERP mutations in promoters of many GO terms are deleterious - GERP mutations in two or more genomic regions of the GO terms: androgen metabolism, immune response and response to oxidative stress are deleterious - High impact SnpEff mutations in all genomic regions of immune response are negative</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../qmd/5_per_region.html" class="pagination-link" aria-label="Load per genomic region">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Load per genomic region</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../qmd/7_random_sub.html" class="pagination-link" aria-label="Random subsampling">
        <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Random subsampling</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>